pandas/tseries/offsets.py

```diff
@@ -1160,6 +1162,196 @@ def onOffset(self, dt):
     _prefix = 'MS'
 
 
+class SemiMonthOffset(DateOffset):
+    _adjust_dst = True
+    _default_day_of_month = 15
+    _min_day_of_month = 2
+
+    def __init__(self, n=1, normalize=False, **kwds):
+        self.day_of_month = kwds.get('day_of_month',
+                                     self._default_day_of_month)
+
+        if not self._min_day_of_month <= self.day_of_month <= 27:
+            raise ValueError('day_of_month must be '
+                             '{}<=day_of_month<=27, got {}'.format(
+                                 self._min_day_of_month, self.day_of_month))
+        self.n = int(n)
+        self.normalize = normalize
+        self.kwds = kwds
+
+    @classmethod
+    def _from_name(cls, suffix=None):
+        if not suffix:
```

Comment:
`._from_name`  is used here: https://github.com/pydata/pandas/blob/master/pandas/tseries/frequencies.py#L566 

without it you get a value error when calling `.to_offset`
```bash
In [4]: pd.tseries.frequencies.to_offset('SMS')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/Users/adrienemery/Code/pandas/pandas/tseries/frequencies.py in to_offset(freqstr)
    464             for stride, name, _ in opattern.findall(freqstr):
--> 465                 offset = get_offset(name)
    466                 if stride_sign is None:

/Users/adrienemery/Code/pandas/pandas/tseries/frequencies.py in get_offset(name)
    565             # many '-')
--> 566             offset = klass._from_name(*split[1:])
    567         except (ValueError, TypeError, KeyError):

AttributeError: type object 'SemiMonthBegin' has no attribute '_from_name'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-4-add2ac572fb8> in <module>()
----> 1 pd.tseries.frequencies.to_offset('SMS')

/Users/adrienemery/Code/pandas/pandas/tseries/frequencies.py in to_offset(freqstr)
    475                     delta = delta + offset
    476         except Exception:
--> 477             raise ValueError("Could not evaluate %s" % freqstr)
    478
    479     if delta is None:

ValueError: Could not evaluate SMS
```